build/patches/Timezone-Override.patch

```diff
@@ -0,0 +1,1386 @@
+@@ -1428,3 +1428,19 @@ static jboolean JNI_WebsitePreferenceBridge_GetMicManagedByCustodian(
+   return IsContentSettingManagedByCustodian(
+       jbrowser_context_handle, ContentSettingsType::MEDIASTREAM_MIC);
+ }
++
++static void JNI_WebsitePreferenceBridge_SetCustomTimezone(
++    JNIEnv* env,
++    const JavaParamRef<jobject>& jbrowser_context_handle,
++    const JavaParamRef<jstring>& timezone) { 
++  std::string new_timezone = ConvertJavaStringToUTF8(env, timezone);
++  GetHostContentSettingsMap(jbrowser_context_handle)->SetTimezoneOverrideValue(new_timezone);
++}
++
++static base::android::ScopedJavaLocalRef<jstring> JNI_WebsitePreferenceBridge_GetCustomTimezone(
++    JNIEnv* env,
++    const JavaParamRef<jobject>& jbrowser_context_handle) {
++  std::string timezone;
++  GetHostContentSettingsMap(jbrowser_context_handle)->GetTimezoneOverrideValue(timezone);
++  return ConvertUTF8ToJavaString(env, timezone);
++}
+diff --git a/components/browser_ui/strings/android/site_settings.grdp b/components/browser_ui/strings/android/site_settings.grdp
+--- a/components/browser_ui/strings/android/site_settings.grdp
++++ b/components/browser_ui/strings/android/site_settings.grdp
+@@ -66,6 +66,9 @@
+   <message name="IDS_VR_PERMISSION_TITLE" desc="Title of the permission to use Virtual Reality [CHAR-LIMIT=32]">
+     Virtual reality
+   </message>
++  <message name="IDS_TIMEZONE_OVERRIDE_PERMISSION_TITLE" desc="Title of the permission to use TimeZone Override [CHAR-LIMIT=32]">
++    Timezone override
++  </message>
+ 
+   <!-- Site settings global toggles -->
+ 
+@@ -462,6 +465,38 @@
+     Block sites from playing protected content
+   </message>
+ 
++  <!-- Timezone override (bromite #331) -->
++  <message name="IDS_WEBSITE_SETTINGS_TIMEOVERRIDE_INFO" desc="Primary text explaining system timezone feature.">
++    bla bla bla
++  </message>
++  <message name="IDS_WEBSITE_SETTINGS_CATEGORY_TIMEZONE_OVERRIDE_ALLOWED_TEXT" desc="Primary text explaining that sites are allowed to access system timezone.">
++    Use System Timezone
++  </message>
++  <message name="IDS_WEBSITE_SETTINGS_CATEGORY_TIMEZONE_OVERRIDE_RANDOM_TEXT" desc="Primary text for random timezone.">
++    Random Value
++  </message>
++  <message name="IDS_WEBSITE_SETTINGS_CATEGORY_TIMEZONE_OVERRIDE_ALLOWED" desc="Summary text explaining that sites are allowed to access system timezone.">
++    Allow sites to use System Timezone
++  </message>
++  <message name="IDS_WEBSITE_SETTINGS_CATEGORY_TIMEZONE_OVERRIDE_CUSTOM" desc="Summary text explaining that sites use custom timezone.">
++    Use custom timezone
++  </message>
++  <message name="IDS_WEBSITE_SETTINGS_CATEGORY_TIMEZONE_OVERRIDE_CUSTOM_HINT" desc="Hint text for edit custom timezone.">
++    Default UTC
++  </message>
++  <message name="IDS_WEBSITE_SETTINGS_CATEGORY_TIMEZONE_OVERRIDE_RANDOM" desc="Summary text for random timezone.">
++    Use random timezone for each page
```

Comment:
```suggestion
+    Use a random timezone for each tab
```
@uazo is it for each page or for each tab?

Reply:
@csagan5 
**per page**, my definitive answer.
if it is true that it is called by **AllowScript()**, which is checked for each origin, we change the offset to each page and iframe.

try https://uazo.github.io/bromite/test.html 
if you make a similar one, I'll put an `<A>` with that url, so we verify

Reply:
It would be best that it would not change for iframes, some webpages would show Javascript-generated content in different timezones this way.

Reply:
iframe **can be** a RenderFrame.
I don't know for sure. I think site-isolation has something to do with it, but I haven't checked.
however it would not be easy I think, also because there could be a security problem there, because somehow the iframe should access the information in the parent.
I'd say leave it like this, let's see if any bugs come out.